lib/chef/mixin/user_context.rb

```diff
@@ -0,0 +1,52 @@
+#
+# Author:: Adam Edwards (<adamed@chef.io>)
```

Comment:
@stevendanna  let me know if you'd like my help moving this forward. If the issue is that we need an end to end test, I can help generate one, though honestly, the functional tests *do* actually verify that it works. 😄